src/lib/connectors/index.ts

```diff
@@ -14,6 +15,8 @@ export const newConnector = (connection: Connection): Connector => {
   switch (connection.engineType) {
     case Engine.MySQL:
       return mysql(connection);
+    case Engine.SingleStore:
+      return singlestore(connection);
```

Comment:
The singlestore connector seems to be identical (or almost) to the MySQL one, thus there is no need to duplicate the code.

We can use the same mysql connector. And if there is some divergence, we can pass in the engineType into the connector and use `if (engineType == Engine.SingleStore)`